utils/db-generator/query_db_generator.cpp

```diff
@@ -1264,8 +1269,41 @@ TableList getTablesFromSelect(std::vector<std::string> queries)
     return result;
 }
 
-int main(int, char **)
+int main(int argc, const char *argv[])
 {
+    try
+    {
+        po::options_description desc("Allowed options");
+        desc.add_options()
+            ("help,h", "Display greeting and allowed options.")
+            ("input,i", po::value<std::string>(), "Input filename.")
+            ("output,o", po::value<std::string>(), "Output filename.");
+
+        po::variables_map vm;
+        po::store(po::parse_command_line(argc, argv, desc), vm);
+        po::notify(vm);
+
+        if (vm.count("help") || vm.count("h"))
+        {
+            std::cout << "Hello! It is datasets generator for ClickHouse's queries." << std::endl;
+            std::cout << "Put some query as an input and it will produce queries for table creating and filling." << std::endl;
+            std::cout << "After that your query could be executed on this tables." << std::endl;
+            std::cout << desc << std::endl;
+            return 1;
+        }
+        if (vm.count("input"))
+            freopen(vm["input"].as<std::string>().c_str(), "r", stdin);
```

Comment:
BTW, I tried with stdin but it didn't work either.

Reply:
Maybe there were no semicolon at the end of query. Or you didn't press Ctrl-d..

```
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ cat input.sql 
SELECT a from b;
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ ./query_db_generator -i input.sql
CREATE DATABASE IF NOT EXISTS ;

CREATE TABLE IF NOT EXISTS b (
a Int64
) ENGINE = Log;

INSERT INTO b
(a) VALUES 
(495582894);

SELECT a from b; 
☁️ ☁️ ☁️ @jakalletti-build:~/ClickHouse/build/memory/utils/db-generator  
↳ $ ./query_db_generator 
Copy your queries (with semicolons) here, press Enter and Ctrl+D.
SELECT a from b;
CREATE DATABASE IF NOT EXISTS ;

CREATE TABLE IF NOT EXISTS b (
a Int64
) ENGINE = Log;

INSERT INTO b
(a) VALUES 
(495582894);

SELECT a from b; 

```